workflows/prognostic_c48_run/runtime/steppers/prescriber.py

```diff
@@ -0,0 +1,274 @@
+            Hashable, Hashable
+        ] = config.rename if config.rename else {}
+        self._timesteps = timesteps
+        self._prescribed_ds: xr.Dataset = self._scatter_prescribed_ds()
+
+    def _scatter_prescribed_ds(self):
+        time_coord: Optional[xr.DataArray]
+        if self._communicator.rank == 0:
+            prescribed_ds, time_coord = _get_prescribed_ds(
+                self._config.dataset_key,
+                list(self._config.variables),
+                self._timesteps,
+                self._config.consolidated,
+            )
+            prescribed_ds = _quantity_state_to_ds(
+                self._communicator.scatter_state(_ds_to_quantity_state(prescribed_ds))
+            )
+        else:
+            prescribed_ds = _quantity_state_to_ds(self._communicator.scatter_state())
+            time_coord = None
+        time_coord = self._communicator.comm.bcast(time_coord, root=0)
+        prescribed_ds = prescribed_ds.assign_coords({"time": time_coord})
+        return prescribed_ds
+
+    def __call__(self, time, state):
+        diagnostics: Diagnostics = {}
+        prescribed_timestep: xr.Dataset = _cast_to_double(
+            _add_net_shortwave(self._prescribed_ds.sel(time=time))
+        )
+        state_updates: State = self._rename_state(
+            {name: prescribed_timestep[name] for name in prescribed_timestep.data_vars}
+        )
+        for name in state_updates.keys():
+            diagnostics[name] = state_updates[name]
+        tendency: Tendencies = {}
+        return tendency, diagnostics, state_updates
+
+    def _rename_state(self, state: State) -> State:
+        new_state: State = {}
+        for name in state.keys():
+            if name in self._rename.keys():
+                new_state[self._rename[name]] = state[name]
+            else:
+                new_state[name] = state[name]
+        return new_state
+
+    def get_diagnostics(self, state, tendency):
+        return {}
+
+    def get_momentum_diagnostics(self, state, tendency):
+        return {}
+
+
+def _get_prescribed_ds(
+    dataset_key: str,
+    variables: Sequence[str],
+    timesteps: Optional[Sequence[cftime.DatetimeJulian]],
+    consolidated: bool = True,
+) -> Tuple[xr.Dataset, xr.DataArray]:
+    logger.info(f"Setting up dataset for state setting: {dataset_key}")
+    ds = _open_ds(dataset_key, consolidated)
+    ds = _remove_name_suffix(ds)
+    ds = get_variables(ds, variables)
+    ds = _rename_dims(ds)
+    ds = _set_missing_units_attr(ds)
+    ds = _round_time_coord(ds)
+    if timesteps is not None:
+        ds = ds.sel(time=timesteps)
+    time_coord = ds.coords["time"]
+    return ds.drop_vars(names="time").load(), time_coord
+
+
+def _open_ds(dataset_key: str, consolidated: bool) -> xr.Dataset:
+    try:
+        zarr_path = CATALOG[dataset_key].urlpath
+    except KeyError:
+        zarr_path = dataset_key
+    return intake.open_zarr(zarr_path, consolidated=consolidated).to_dask()
```

Comment:
Is there a reason to use `intake.open_zarr` instead of `xr.open_zarr` here?

I think the typical pattern for opening a dataset with intake would be to do:

`ds = CATALOG[dataset_key].to_dask()`

which avoids grabbing the `urlpath` or baking in `zarr` format assumption in the code.

Reply:
My reason for not doing this is that the catalog approach doesn't allow for opening with `consolidated=True` (as far as I can tell?)

Reply:
Got it. The catalog entries do contain a `consolidated` boolean as metadata, so I'd hope intake would use that, but I'm not sure TBH.

Reply:
Oh OK, will just let intake do that then.